### DISCUSSION TITLE
Add watermark to images

### DISCUSSION THREAD

**Discussion by kimlundjohansen** [PiranhaCMS/piranha.core#2057] [Q&A] [answered]
Is it possible through an extension or hook to add a watermark to a picture uploaded in the media section of the manager?

**Comment by tidyui** [comment, marked as answer]
Hi there! You should be able to use the following hook to process any media after it has been saved.

~~~csharp
App.Hooks.Media.RegisterOnAfterSave((media) => 
{
   // Add custom code to process the binary file if it has been updated
});
~~~

You find documentation on all of the available hooks here:

https://piranhacms.org/docs/master/application/hooks

**Reply by kimlundjohansen** [reply]
Nice, thanks :-)
